File: src/Daemon/AlertMessage.cs
using ServerCheck.Config;
using ServerCheck.Sender;

namespace ServerCheck.Daemon
{
    class AlertMessage
    {
        private static readonly string _errorMessage = ConfigManager.GetAppSetting("errorMessage");
        private static readonly string _solveMessage = ConfigManager.GetAppSetting("solveMessage");

        private readonly bool _outcome;
        private readonly string _serverAddress;
        private readonly string _checkTimestamp;

        private string _text = "";

        public AlertMessage(bool outcome, string server_address, string check_timestamp)
        {
            this._outcome = outcome;
            this._serverAddress = server_address;
            this._checkTimestamp = check_timestamp;

            this.GenerateText();
        }

        private void GenerateText()
        {
            this._text = this._outcome ? AlertMessage._solveMessage : AlertMessage._errorMessage;
            this._text = this._text.Replace("[SERVER_ADDRESS]", this._serverAddress).Replace("[DATE]", this._checkTimestamp);
        }

        public void Notify(string to, string contact_type)
        {
            ISender sender = SenderCreator.GetSender(contact_type);
            sender.Send(to, this._text);
        }
    }
} 

File: app/src/main/java/com/antonioscardace/app/Daemon/AlertMessage.java
package com.antonioscardace.app.Daemon;

import java.io.IOException;

import javax.mail.MessagingException;

import com.antonioscardace.app.Config;
import com.antonioscardace.app.Sender.ISender;
import com.antonioscardace.app.Sender.SenderCreator;

public final class AlertMessage {
    private final String errorMessage;
    private final String solveMessage;

    private final boolean outcome;
    private final String serverAddress;
    private final String checkTimestamp;

    public AlertMessage(boolean outcome, String server, String timestamp) {
        this.outcome = outcome;
        this.serverAddress = server;
        this.checkTimestamp = timestamp;

        this.solveMessage = Config.getInstance().get("solveMessage");
        this.errorMessage = Config.getInstance().get("errorMessage");
    }

    private String generateText() {
        String text = this.outcome ? this.solveMessage : this.errorMessage;
        return text.replace("[SERVER_ADDRESS]", this.serverAddress)
                .replace("[DATE]", this.checkTimestamp)
                .replace("[HI]", "👋🏻")
                .replace("[ALERT]", "⚠️")
                .replace("[OK]", "✅");
    }

    public void notify(String to, String contactType) throws IOException, MessagingException {
        ISender sender = SenderCreator.getSender(contactType);
        String message = this.generateText();
        sender.send(to, message);
    }
}